--- src/app/api/image/route.ts
+++ src/app/api/image/route.ts
@@ -62,11 +62,14 @@
     }
 
+    const responseHeaders = new Headers({
+      'Content-Type': response.headers.get('Content-Type') || 'image/jpeg',
+      'Cache-Control': 'public, max-age=31536000, immutable',
+      'Access-Control-Allow-Origin': '*'
+    });
+    if (response.headers.has('cf-resized')) {
+      responseHeaders.set('cf-resized', response.headers.get('cf-resized') as string);
+    }
+
     // Construct the response to cache
     const finalResponse = new Response(response.body, {
       status: response.status,
       statusText: response.statusText,
-      headers: new Headers({
-        'Content-Type': response.headers.get('Content-Type') || 'image/jpeg',
-        'Cache-Control': 'public, max-age=31536000, immutable',
-        'Access-Control-Allow-Origin': '*'
-      })
+      headers: responseHeaders
     });
